StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1518UnitTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using StyleCop.Analyzers.LayoutRules;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using TestHelper;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace StyleCop.Analyzers.Test.LayoutRules
+{
+    /// <summary>
+    /// Unit tests for the SA1518: CodeMustNotContainBlankLinesAtEndOfFile
+    /// </summary>
+    [TestClass]
+    public class SA1518UnitTests : CodeFixVerifier
+    {
+        private const string DiagnosticId = SA1518CodeMustNotContainBlankLinesAtEndOfFile.DiagnosticId;
+
+        private const string BaseCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+        Debug.Assert(true);
+    }
+}";
+        
+        /// <summary>
+        /// Verifies that the analyzer will properly handle an empty source.
+        /// </summary>
+        /// <returns></returns>
+        [TestMethod, TestCategory("MaintainabilityRules/SA1518")]
```

Comment:
Since the tests are all placed in classes according to the name of the tested diagnostic, I don't think it makes sense to add this test category at this time. You can use Test Explorer to group tests according to class.

![image](https://cloud.githubusercontent.com/assets/1408396/6407966/b4a8080e-be10-11e4-817c-47fe4eb9a229.png)